Mathlib/Algebra/Module/Injective.lean

```diff
@@ -0,0 +1,467 @@
+set_option linter.uppercaseLean3 false in
+#align module.Baer.extension_of_max_adjoin.extension_to_fun Module.Baer.ExtensionOfMaxAdjoin.extensionToFun
+
+set_option synthInstance.etaExperiment true in
+theorem ExtensionOfMaxAdjoin.extensionToFun_wd (h : Module.Baer R Q) {y : N}
+    (x : supExtensionOfMaxSingleton i f y) (a : (extensionOfMax i f).domain)
+    (r : R) (eq1 : ↑x = ↑a + r • y) :
+    ExtensionOfMaxAdjoin.extensionToFun i f h x =
+      (extensionOfMax i f).toLinearPMap a + ExtensionOfMaxAdjoin.extendIdealTo i f h y r := by
+  cases' a with a ha
+  have eq2 :
+    (ExtensionOfMaxAdjoin.fst i x - a : N) = (r - ExtensionOfMaxAdjoin.snd i x) • y := by
+    change x = a + r • y at eq1
+    rwa [ExtensionOfMaxAdjoin.eqn, ← sub_eq_zero, ← sub_sub_sub_eq, sub_eq_zero, ← sub_smul]
+      at eq1
+  have eq3 :=
+    ExtensionOfMaxAdjoin.extendIdealTo_eq i f h (r - ExtensionOfMaxAdjoin.snd i x)
+      (by rw [← eq2]; exact Submodule.sub_mem _ (ExtensionOfMaxAdjoin.fst i x).2 ha)
+  simp only [map_sub, sub_smul, sub_eq_iff_eq_add] at eq3
+  unfold ExtensionOfMaxAdjoin.extensionToFun
+  rw [eq3, ← add_assoc, ← (extensionOfMax i f).toLinearPMap.map_add, AddMemClass.mk_add_mk]
+  congr
+  ext
+  dsimp
+  rw [Subtype.coe_mk, add_sub, ← eq1]
+  exact eq_sub_of_add_eq (ExtensionOfMaxAdjoin.eqn i x).symm
+set_option linter.uppercaseLean3 false in
+#align module.Baer.extension_of_max_adjoin.extension_to_fun_wd Module.Baer.ExtensionOfMaxAdjoin.extensionToFun_wd
+
+set_option synthInstance.etaExperiment true in
+/-- The linear map `M ⊔ ⟨y⟩ ⟶ Q` by `x + r • y ↦ f x + φ r` is an extension of `f`-/
+def extensionOfMaxAdjoin (h : Module.Baer R Q) (y : N) : ExtensionOf i f where
+  domain := supExtensionOfMaxSingleton i f y -- (extensionOfMax i f).domain ⊔ Submodule.span R {y}
+  le := le_trans (extensionOfMax i f).le le_sup_left
+  toFun :=
+    { toFun := ExtensionOfMaxAdjoin.extensionToFun i f h
+      map_add' := fun a b => by
+        have eq1 :
+          ↑a + ↑b =
+            ↑(ExtensionOfMaxAdjoin.fst i a + ExtensionOfMaxAdjoin.fst i b) +
+              (ExtensionOfMaxAdjoin.snd i a + ExtensionOfMaxAdjoin.snd i b) • y := by
+          rw [ExtensionOfMaxAdjoin.eqn, ExtensionOfMaxAdjoin.eqn, add_smul, Submodule.coe_add]
+          ac_rfl
+        rw [ExtensionOfMaxAdjoin.extensionToFun_wd (y := y) i f h (a + b) _ _ eq1,
+          LinearPMap.map_add, map_add]
+        unfold ExtensionOfMaxAdjoin.extensionToFun
+        abel
+      map_smul' := fun r a => by
+        dsimp
+        have eq1 :
+          r • (a:N) =
+            ↑(r • ExtensionOfMaxAdjoin.fst i a) + (r • ExtensionOfMaxAdjoin.snd i a) • y := by
+          rw [ExtensionOfMaxAdjoin.eqn, smul_add, smul_eq_mul, mul_smul]
+          rfl
+        rw [ExtensionOfMaxAdjoin.extensionToFun_wd i f h (r • a) _ _ eq1, LinearMap.map_smul,
+          LinearPMap.map_smul, ← smul_add]
+        congr }
+  is_extension m := by
+    dsimp
```

Comment:
Do you understand why this is needed in mathlib4 but not mathlib3? Is it worth a porting note?

Reply:
I think is the change in default transparency.